Hw.rb

```diff
@@ -0,0 +1,206 @@
+students = [
+  {
+      first_name: "Wejdan",
+      last_name: " Alghamdi"
+  },
+  {
+      first_name: "Shahad",
+      last_name: " Bagarish",
+  },
+  {
+      first_name: "Doaa",
+      last_name: " Alalawi",
+  }
+]
+upper_case_full_names = []
+for  i  in students  do
+    fullname=i[:first_name] + i[:last_name]
+    fullname= fullname.upcase
+    upper_case_full_names.push(fullname)
+end
+
+puts upper_case_full_names.inspect
+
+
+
+#-------------------------------
+
+
+users = [
+  {
+      name: "Mohammad",
+      orders: [
+          {
+              description: "a bmx"
+          }
+      ]
+  },
+  {
+      name: "Ibtissam",
+      orders: [
+          {
+              description: "honey"
+          },
+          {
+              description: "Nintendo 3DS"
+          }
+      ]
+  },
+  {
+      name: "Yazeed",
+      orders: [
+          {
+              description: "a Razer Blade"
+          },
+          {
+              description: "The Arrow DVDs"
+          },
+          {
+              description: "airpods"
+          },
+          {
+              description: "a unicorn"
+          }
+      ]
+  }
+]
+first_order_for_each_user = []
+
+
+for i in users do 
+
+    first_order_for_each_user.push(i[:orders][0])
+end 
+puts first_order_for_each_user
+
+#----------------------------------------------
+
+people = [
+  {
+      name: "Ashjan",
+      transactions: [
+          {
+              type: "COFFEE",
+              amount: 7.43
+          },
+          {
+              type: "TACOS",
+              amount: 14.65
+          },
+          {
+              type: "COFFEE",
+              amount: 4.43
+          }
+      ]
+  },
+  {
+      name: "Abeer",
+      transactions: [
+          {
+              type: "BIKES",
+              amount: 800.00
+          },
+          {
+              type: "TACOS",
+              amount: 14.65
+          },
+          {
+              type: "COFFEE",
+              amount: 4.43
+          }
+      ]
+  },
+  {
+      name: "Nujud",
+      transactions: [
+          {
+              type: "COFFEE",
+              amount: 7.43
+          },
+          {
+              type: "COFFEE",
+              amount: 100.00
+          },
+          {
+              type: "COFFEE",
+              amount: 4.43
+          }
+      ]
+  }
+]
+coffee_average_per_person = []
+sum=0
+count = 0
+for i in people do 
+    for j in i[:transactions]
+    if i[:type]=="COFFEE"
+        sum=sum+i[:amount]
+count++
+  
+    end
+end
+    avrge=sum/count
+    coffee_average_per_person.push("name = " + i[:name] + " the amount is "+ avrge )
```

Comment:
the variable``` avrge``` is integer and in ruby **you can not !**   sum between integer and string like this ! 
if you want sum between integer and string do that 
``` " the amount is "+ avrge.to_s ```